Add Rehlko (formerly Kohler Energy Management) integration (#38782)
home-assistant/home-assistant.github.io: source/_integrations/rehlko.markdown

---
title: Rehlko

> description: Documentation about the Rehlko devices.

Reviewer: ```suggestion
description: Support for Rehlko devices (formerly Oncue and Kohler Energy)
```

ha_category:
  - Sensor
ha_iot_class: Cloud Polling
ha_release: 2025.5
ha_config_flow: true
ha_codeowners:
  - '@bdraco'
  - '@peterager'
ha_dhcp: true
ha_domain: rehlko
ha_platforms:
  - sensor
ha_integration_type: integration
---

The Rehlko (formerly Kohler Energy) integration monitors the state of your [Enabled Kohler generator](https://www.kohlerhomeenergy.rehlko.com/kohler-energy-management-app). This integration replaces the Oncue integration.

> To configure the integration you will need your log in credentials (email and password) for the Kohler app.

Reviewer: ```suggestion
To configure the integration you will need your log in credentials (email and password) form the manufacturers app.
```

---
title: Rehlko
description: Support for Rehlko devices (formerly Oncue and Kohler Energy)
ha_category:
  - Sensor
ha_iot_class: Cloud Polling
ha_release: 2025.5
ha_config_flow: true
ha_codeowners:
  - '@bdraco'
  - '@peterager'
ha_dhcp: true
ha_domain: rehlko
ha_platforms:
  - sensor
ha_integration_type: integration
---

The Rehlko integration allows you to monitor the status of your [enabled Kohler generator](https://www.kohlerhomeenergy.rehlko.com/kohler-energy-management-app) through Home Assistant. This integration replaces the legacy Oncue integration.

### Configuration

To set up the integration, you’ll need your login credentials (email and password) for the manufacturer's app.

{% include integrations/config_flow.md %}

### Sensor Availability

The sensors exposed by this integration depend on the specific instrumentation installed on your generator and transfer switch. Sensor availability can vary significantly—even between units of the same model.

## Tested generators

> There are the geneator models that have been tested.

Reviewer: ```suggestion
These are the generator models that have been tested:
```

- [38RCLB](https://resources.kohler.com/power/kohler/residential/pdf/tp6908.pdf)
- [20RESA](https://resources.kohler.com/power/kohler/residential/pdf/tp6804.pdf)
- [20RCA](https://www.kohlerhomeenergy.rehlko.com/products/home+generators/20rca)
- [14RESA](https://www.kohler.com/content/dam/kohler-com-NA/Lifestyle/PDF/PDF-tp6803.pdf)

> {% include integrations/config_flow.md %}

Reviewer: ```suggestion
```

---
title: Rehlko
description: Support for Rehlko devices (formerly Oncue and Kohler Energy)
ha_category:
  - Sensor
ha_iot_class: Cloud Polling
ha_release: 2025.5
ha_config_flow: true
ha_codeowners:
  - '@bdraco'
  - '@peterager'
ha_dhcp: true
ha_domain: rehlko
ha_platforms:
  - sensor
ha_integration_type: integration
---

> The Rehlko (formerly Kohler Energy Management) integration allows you to monitor the status of your [enabled Kohler generator](https://www.kohlerhomeenergy.rehlko.com/kohler-energy-management-app) through Home Assistant. This integration replaces the legacy Oncue integration.

Reviewer 2: ```suggestion
The **Rehlko** (formerly Kohler Energy Management) {% term integration %} allows you to monitor the status of your [enabled Kohler generator](https://www.kohlerhomeenergy.rehlko.com/kohler-energy-management-app) through Home Assistant. This integration replaces the legacy Oncue integration.
```

{% include integrations/config_flow.md %}

> ### Required manual input

Reviewer 2: ```suggestion
## Required manual input
```

To configure a Rehlko (Kohler Energy Management) account, enter the following information:

{% configuration_basic %}

email:
  description: "The email used to log in to the Rehlko (Kohler Energy Management) application."
password:
  description: "The password used to log in to the Rehlko (Kohler Energy Management) application."

{% endconfiguration_basic %}

> ## Removing the integration
> 
> This integration follows the standard integration removal process; no extra steps are required.
> 
> {% include integrations/remove_device_service.md %}
> 

Reviewer 2: ```suggestion
```

> ## Sensor Availability

Reviewer 2: ```suggestion
## Sensor availability
```

> ### Sensor Availability

Reviewer: ```suggestion
## Sensor Availability
```

The sensors exposed by this integration depend on the specific instrumentation installed on your generator and transfer switch. Sensor availability can vary significantly—even between units of the same model.

## Tested generators

These are the generator models that have been tested:

- [38RCLB](https://resources.kohler.com/power/kohler/residential/pdf/tp6908.pdf)
- [20RESA](https://resources.kohler.com/power/kohler/residential/pdf/tp6804.pdf)
- [20RCA](https://www.kohlerhomeenergy.rehlko.com/products/home+generators/20rca)
- [14RESA](https://www.kohler.com/content/dam/kohler-com-NA/Lifestyle/PDF/PDF-tp6803.pdf)

> 

Reviewer 2: ```suggestion

## Removing the integration

This integration follows the standard integration removal process; no extra steps are required.

{% include integrations/remove_device_service.md %}

```

---
title: Rehlko
description: Support for Rehlko devices (formerly Oncue and Kohler Energy)
ha_category:
  - Sensor
ha_iot_class: Cloud Polling
ha_release: 2025.5
ha_config_flow: true
ha_codeowners:
  - '@bdraco'
  - '@peterager'
ha_dhcp: true
ha_domain: rehlko
ha_platforms:
  - sensor
ha_integration_type: integration
---

The Rehlko (formerly Kohler Energy Management) integration allows you to monitor the status of your [enabled Kohler generator](https://www.kohlerhomeenergy.rehlko.com/kohler-energy-management-app) through Home Assistant. This integration replaces the legacy Oncue integration.

> ### Configuration
> 
> To set up the integration, you’ll need your login credentials (email and password) for the manufacturer's app.
> 

Reviewer: ```suggestion
```
